src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/moe_matmuls_fusion.cpp

```diff
@@ -87,9 +88,14 @@ ov::intel_cpu::MoE2GeMMFusion::MoE2GeMMFusion() {
         pattern::wrap_type<ov::op::v1::Minimum>({slice2, pattern::wrap_const()}, pattern::consumers_count(1));
     auto swish_beta = pattern::wrap_const();
     auto swish = pattern::wrap_type<ov::op::v4::Swish>({minimum1, swish_beta}, pattern::consumers_count(1));
+    auto mul1_const_predicate = [](const ov::Output<ov::Node>& output) -> bool {
+        return pattern::rank_equals(3)(output) && output.get_shape()[1] == 1;
+    };
+    auto mul1_const = pattern::wrap_type<ov::op::v0::Constant>(mul1_const_predicate);
```

Comment:
Let's use `pattern::shape_matches` instead of a custom predicate:
```suggestion
    auto mul1_const = pattern::wrap_type<ov::op::v0::Constant>(pattern::shape_matches("[?, 1, ?]");
```